src/rendering/src/components/Navigation/HeaderContent.tsx

```diff
@@ -29,23 +29,23 @@ const HeaderContent = (props: HeaderContentProps): JSX.Element => {
           query: props.query,
         },
         props.asPath,
-        { locale: lang }
+        {
+          locale: lang,
+          shallow: false,
+        }
       );
     }
   };
 
   const languageSelector = languageList && (
-    <select
-      onChange={(e) => changeLanguage(e.currentTarget.value)}
-      className="languagePicker"
-      defaultValue={sitecoreContext.language}
-    >
+    <select onChange={(e) => changeLanguage(e.currentTarget.value)} className="languagePicker">
       {languageList.map((language, index) => (
         <option
           key={index}
           value={language['Name']}
           label={languageNames.of(language['Name'])}
           className="languageItem"
+          selected={sitecoreContext.language === language['Name']}
```

Comment:
With this change, React throws an error in the browser console that says: Warning: Use the `defaultValue` or `value` props on <select> instead of setting `selected` on <option>.